File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> elem = new HashSet();

        for(int i=0;i<nums.length;i++){
            elem.add(nums[i]);
        }
        
        int c = 0;
        int len= 0;
        for(int i : elem) {
            int n = i;
            if(!elem.contains(i-1)){
                n++;
                c = 1;
                while(elem.contains(n)){
                    n++;
                    c++;
                }
                len = Math.max(len, c);
            }
        }

        return len;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)

        length = 0

        for num in nums_set:
            temp = num
            if temp-1 not in nums_set:
                temp_length = 0
                while temp in nums_set:
                    temp_length += 1
                    temp += 1
                length = max(length, temp_length)

        return length

